core/src/main/java/org/apache/iceberg/view/ViewVersionParser.java

```diff
@@ -0,0 +1,112 @@
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.util.JsonUtil;
+
+class ViewVersionParser {
+
+  private static final String VERSION_ID = "version-id";
+  private static final String TIMESTAMP_MS = "timestamp-ms";
+  private static final String SUMMARY = "summary";
+  private static final String OPERATION = "operation";
+
+  private static final String REPRESENTATIONS = "representations";
+
+  private ViewVersionParser() {}
+
+  static void toJson(ViewVersion version, JsonGenerator generator) throws IOException {
+    Preconditions.checkArgument(version != null, "Invalid view version: null");
+    generator.writeStartObject();
+    generator.writeNumberField(VERSION_ID, version.versionId());
+    generator.writeNumberField(TIMESTAMP_MS, version.timestampMillis());
+    generator.writeObjectFieldStart(SUMMARY);
+    generator.writeStringField(OPERATION, version.operation().name().toLowerCase(Locale.ENGLISH));
+    for (Map.Entry<String, String> summaryEntry : version.summary().entrySet()) {
+      if (!summaryEntry.getKey().equals(OPERATION)) {
+        generator.writeStringField(summaryEntry.getKey(), summaryEntry.getValue());
+      }
+    }
+
+    generator.writeEndObject();
+
+    generator.writeArrayFieldStart(REPRESENTATIONS);
+    for (ViewRepresentation representation : version.representations()) {
+      ViewRepresentationParser.toJson(representation, generator);
+    }
+    generator.writeEndArray();
+
+    generator.writeEndObject();
+  }
+
+  static String toJson(ViewVersion version) {
+    return JsonUtil.generate(gen -> toJson(version, gen), false);
+  }
+
+  static ViewVersion fromJson(String json) {
+    return JsonUtil.parse(json, ViewVersionParser::fromJson);
+  }
+
+  static ViewVersion fromJson(JsonNode node) {
+    Preconditions.checkArgument(node != null, "Cannot parse view version from null object");
+
+    Preconditions.checkArgument(
+        node.isObject(), "Cannot parse view version from a non-object: %s", node);
+
+    int versionId = JsonUtil.getInt(VERSION_ID, node);
+    long timestamp = JsonUtil.getLong(TIMESTAMP_MS, node);
+    Map<String, String> summary = Maps.newHashMap();
+    Map<String, String> serializedSummary = JsonUtil.getStringMap(SUMMARY, node);
+    ViewVersion.Operation operation = null;
+
+    for (Map.Entry<String, String> summaryEntry : serializedSummary.entrySet()) {
+      if (summaryEntry.getKey().equals(OPERATION)) {
+        operation = ViewVersion.Operation.fromName(summaryEntry.getValue());
+      } else {
+        summary.put(summaryEntry.getKey(), summaryEntry.getValue());
+      }
+    }
```

Comment:
When deserializing the operation we take it out of the summary map, that way the summary map has everything else, and we set the operation on the version itself. Same as `SnapshotParser` and operation